### PR TITLE
Fix in extra_coords

### PR DESCRIPTION
**Pull request by aurghs**
This PR fixes two bugs on the `extra_coords` keyword:

1) Currently it is not possible to use `extra_coords` with the new cfgrib-xarray interface (`xarray_plugin.py`). I have added `extra_coords` in the new interface to fix it.

2) If the user tries to read an extra coordinate with a scalar dimension,  cfrgib returns an empty coordinate and xarray raise an error because the coordinates and dimensions are not consistent. 

**old behaviour:**
```python
res = xr.open_dataset(
        path, extra_coords={"experimentVersionNumber": "time"}
    )

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/var/folders/44/qn_zhjz165gflnp_1_x0p9g40000gn/T/ipykernel_89177/3004447257.py in <module>
      1 TEST_DATA_SCALAR_TIME = "/Users/barghini/devel/wp6/cfgrib/tests/sample-data/era5-single-level-scalar-time.grib"
      2 
----> 3 res = xr.open_dataset(
      4         path, extra_coords={"experimentVersionNumber": "time"}
      5     )

/usr/local/Caskroom/miniconda/base/envs/cfgrib-wp6/lib/python3.9/site-packages/xarray/backends/api.py in open_dataset(filename_or_obj, engine, chunks, cache, decode_cf, mask_and_scale, decode_times, decode_timedelta, use_cftime, concat_characters, decode_coords, drop_variables, backend_kwargs, *args, **kwargs)
    494 
    495     overwrite_encoded_chunks = kwargs.pop("overwrite_encoded_chunks", None)
--> 496     backend_ds = backend.open_dataset(
    497         filename_or_obj,
    498         drop_variables=drop_variables,

~/devel/wp6/cfgrib/cfgrib/xarray_entrypoint.py in open_dataset(self, filename_or_obj, mask_and_scale, decode_times, concat_characters, decode_coords, drop_variables, use_cftime, decode_timedelta, lock, indexpath, filter_by_keys, read_keys, encode_cf, squeeze, time_dims, errors, extra_coords)

/usr/local/Caskroom/miniconda/base/envs/cfgrib-wp6/lib/python3.9/site-packages/xarray/core/dataset.py in __init__(self, data_vars, coords, attrs)
    737             coords = coords.variables
    738 
--> 739         variables, coord_names, dims, indexes, _ = merge_data_and_coords(
    740             data_vars, coords, compat="broadcast_equals"
    741         )

/usr/local/Caskroom/miniconda/base/envs/cfgrib-wp6/lib/python3.9/site-packages/xarray/core/merge.py in merge_data_and_coords(data, coords, compat, join)
    475     explicit_coords = coords.keys()
    476     indexes = dict(_extract_indexes_from_coords(coords))
--> 477     return merge_core(
    478         objects, compat, join, explicit_coords=explicit_coords, indexes=indexes
    479     )

/usr/local/Caskroom/miniconda/base/envs/cfgrib-wp6/lib/python3.9/site-packages/xarray/core/merge.py in merge_core(objects, compat, join, combine_attrs, priority_arg, explicit_coords, indexes, fill_value)
    629     assert_unique_multiindex_level_names(variables)
    630 
--> 631     dims = calculate_dimensions(variables)
    632 
    633     coord_names, noncoord_names = determine_coords(coerced)

/usr/local/Caskroom/miniconda/base/envs/cfgrib-wp6/lib/python3.9/site-packages/xarray/core/dataset.py in calculate_dimensions(variables)
    195         for dim, size in zip(var.dims, var.shape):
    196             if dim in scalar_vars:
--> 197                 raise ValueError(
    198                     f"dimension {dim!r} already exists as a scalar variable"
    199                 )

ValueError: dimension 'time' already exists as a scalar variable
````

**new behaviour:**
```python
res = xr.open_dataset(
        path, extra_coords={"experimentVersionNumber": "time"}
    )
print(res)

<xarray.Dataset>
Dimensions:                  (latitude: 121, longitude: 201)
Coordinates:
    number                   int64 ...
    time                     datetime64[ns] ...
    step                     timedelta64[ns] ...
    surface                  float64 ...
  * latitude                 (latitude) float64 60.0 59.75 59.5 ... 30.25 30.0
  * longitude                (longitude) float64 -10.0 -9.75 -9.5 ... 39.75 40.0
    valid_time               datetime64[ns] ...
    experimentVersionNumber  object ...
Data variables:
    t2m                      object ...
Attributes:
    GRIB_edition:            1
    GRIB_centre:             ecmf
    GRIB_centreDescription:  European Centre for Medium-Range Weather Forecasts
    GRIB_subCentre:          0
    Conventions:             CF-1.7
    institution:             European Centre for Medium-Range Weather Forecasts
    history:                 2021-07-20T11:32 GRIB to CDM+CF via cfgrib-0.9.9...
```

### CODE REVIEW

File: cfgrib/dataset.py [hunk 1]
```diff
@@ -521,13 +521,18 @@ def build_variable_components(
     extra_coords_data: T.Dict[str, T.Dict[str, T.Any]] = {
         coord_name: {} for coord_name in extra_coords
     }
-    for dim in header_dimensions:
-        header_value_index[dim] = {v: i for i, v in enumerate(coord_vars[dim].data.tolist())}
+    extra_dims = list(extra_coords.values())
```
Comment (alexamici) [inline cfgrib/dataset.py:524]: If you use `tuple` here you may remove the cast to `list` later. 

```suggestion
    extra_dims = tuple(extra_coords.values())
```

File: cfgrib/dataset.py [hunk 2]
```diff
@@ -563,10 +568,14 @@ def build_variable_components(
         coord_vars["valid_time"] = Variable(dimensions=time_dims, data=time_data, attributes=attrs)
 
     for coord_name in extra_coords:
-        coord_vars[coord_name] = Variable(
-            dimensions=(extra_coords[coord_name],),
-            data=np.array(list(extra_coords_data[coord_name].values())),
-        )
+        coord_data = np.array(list(extra_coords_data[coord_name].values()))
```
Comment (alexamici) [inline cfgrib/dataset.py:571]: Why do you need `np.array(list(`? Am I missing something?

```suggestion
        coord_data = np.array(extra_coords_data[coord_name].values())
```

**Review by alexamici — APPROVED** [review]
I suggested a few style fixes. LGTM otherwise.

File: cfgrib/dataset.py (hunk 2 above)
Comment (aurghs) [reply cfgrib/dataset.py:571]: unfortunately yes, I need to apply the list before. Without list, I obtain an array containing only one object dict_values(...) :(

**Comment by alexamici** [comment]
@aurghs this PR also addresses #231 right?

**Pull request merged**